src/xcode/ENA/ENA/Source/Scenes/Events/Store/Model/Checkin.swift

```diff
@@ -0,0 +1,29 @@
+////
+// 🦠 Corona-Warn-App
+//
+
+struct Checkin {
+
+	let id: Int
+	let traceLocationGUID: String
+	let traceLocationVersion: Int
+	let traceLocationType: TraceLocationType
+	let traceLocationDescription: String
+	let traceLocationAddress: String
+	let traceLocationStart: Date
+	let traceLocationEnd: Date
+	let traceLocationDefaultCheckInLengthInMinutes: Int
+	let traceLocationSignature: String
+	let checkinStartDate: Date
+	var checkinEndDate: Date
+	var targetCheckinEndDate: Date
+	let createJournalEntry: Bool
+
+	mutating func update(checkinEndDate: Date) {
+		self.checkinEndDate = checkinEndDate
+	}
+
+	mutating func update(targetCheckinEndDate: Date) {
+		self.targetCheckinEndDate = targetCheckinEndDate
+	}
```

Comment:
I understand that this makes it easier to update this struct in the mock store. But I think it might be clearer to the developer that you always have to go through the store, if this is not available here. I understand that you would then have to create a new instance in the mock store, as we do for the diary mock store, but I think that would be preferable.

Reply:
Yes it results in a lot of boilerplate code. But you point is valid. Changed it.